### ISSUE TITLE
Object-Recognition: Allow score command to take a single file/url or folder

### ISSUE THREAD

**Issue by gjwgit** [mlhubber/mlmodels#47]
$ ml score object-recognition mypic.jpg
$ ml score object-recognition ./images
$ ml score object-recognition http://hounslowurbanfarm.co.uk/wp-content/uploads/2017/03/img-animal-Willow-the-Barn-Owl.jpg

**Comment by simonzhaoms** [comment]
This had been done in the previous commit.
Give it a try :)

**Comment by gjwgit** [comment]
$ ml score object-recognition http://hounslowurbanfarm.co.uk/wp-content/uploads/2017/03/img-animal-Willow-the-Barn-Owl.jpg
mlhub: model 'img-animal-Willow-the-Barn-Owl.jpg' is not installed (/home/gjw/.mlhub/).

Check for the model name amongst those installed:

  $ ml installed

Models can be installed from the ML Hub:

  $ ml install img-animal-Willow-the-Barn-Owl.jpg

Available pakages on the ML Hub can be listed with:

  $ ml available

$ ml
...
This is version 1.4.4 of mlhub.
...

**Comment by simonzhaoms** [comment]
The error cannot be reproduced on my machine.
```console
$ ml score object-recognition http://hounslowurbanfarm.co.uk/wp-content/uploads/2017/03/img-animal-Willow-the-Barn-Owl.jpg
Loading the required Python modules for the ResNet152 model.

Loading the pre-trained ResNet v1 152 model with 1000 classes.
    Model loading time: 5043.38 ms

Recognizing the image
'http://hounslowurbanfarm.co.uk/wp-content/uploads/2017/03/img-animal-Willow-the-Barn-Owl.jpg'
    Computed in 2139.75 ms
    Predictions:
      91.56%: kite
       2.32%: vulture
       0.90%: brambling, Fringilla montifringilla

Thank you for exploring the 'object-recognition' model.

$ ml
ml
Usage: ml [<options>] <command> [<command options>] [<model>]
...
This is version 1.4.4 of mlhub.
...
```

**Comment by gjwgit** [comment]
On a new Deep Learning Data Science Virtual Machine I get the same 

`mlhub: model 'img-animal-Willow-the-Barn-Owl.jpg' is not installed`

```
fred@dl03:~$ pip install mlhub
Collecting mlhub
  Downloading https://files.pythonhosted.org/packages/a0/39/5290aad60934ac7f199ecdcf7225d24f72e78ad6d4288f0c0390e8e20989/mlhub-1.4.5.tar.gz
Requirement already satisfied: requests in /data/anaconda/envs/py35/lib/python3.5/site-packages (from mlhub) (2.18.4)
Requirement already satisfied: pyyaml in /data/anaconda/envs/py35/lib/python3.5/site-packages (from mlhub) (3.12)
Collecting yamlordereddictloader (from mlhub)
  Downloading https://files.pythonhosted.org/packages/56/e1/1ca77da64cc355f0de483095e841d96f2366f93b095b83869440a296c21d/yamlordereddictloader-0.4.0.tar.gz
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /data/anaconda/envs/py35/lib/python3.5/site-packages (from requests->mlhub) (3.0.4)
Requirement already satisfied: certifi>=2017.4.17 in /data/anaconda/envs/py35/lib/python3.5/site-packages (from requests->mlhub) (2018.8.24)
Requirement already satisfied: urllib3<1.23,>=1.21.1 in /data/anaconda/envs/py35/lib/python3.5/site-packages (from requests->mlhub) (1.22)
Requirement already satisfied: idna<2.7,>=2.5 in /data/anaconda/envs/py35/lib/python3.5/site-packages (from requests->mlhub) (2.6)
Building wheels for collected packages: mlhub, yamlordereddictloader
  Running setup.py bdist_wheel for mlhub ... done
  Stored in directory: /home/fred/.cache/pip/wheels/fc/24/6e/38c027cf67dfcce78cccba673645adb65a47685b46d605ee7b
  Running setup.py bdist_wheel for yamlordereddictloader ... done
  Stored in directory: /home/fred/.cache/pip/wheels/cf/bd/88/1f973a38d0574e639260c1f381b04435213bf48076a0f495e8
Successfully built mlhub yamlordereddictloader
Installing collected packages: yamlordereddictloader, mlhub
Successfully installed mlhub-1.4.5 yamlordereddictloader-0.4.0
You are using pip version 18.0, however version 18.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

fred@dl03:~$ ml configure
Executing:  sudo cp bash_completion.d/ml.bash /etc/bash_completion.d
Executing:  ml available > /dev/null
Executing:  ml installed > /dev/null

fred@dl03:~$ bash

fred@dl03:~$ ml available 
The repository 'https://mlhub.ai/' provides the following models:

barchart                 1.1.8  Demonstrate the concept of barcharts.       
beeswarm                 1.0.0  Demonstrate the concept of bee swarm charts.
clothes-recommender      1.1.2  Clothes recommendation using CNN/SVM.       
colorize                 1.1.1  Demonstrate the concept of photo colorizatio...
dogs-cats                1.1.6  Dogs and cats image classification using ker...
iris                     1.1.6  Classical iris plant species classifier.    
movie-recommender        1.1.3  Movie recommendation using the SAR algorthm.
object-recognition       1.2.1  Image object recognition using resnet152.   
ports-plots              1.1.2  Demostrate the concept of visualising data. 
rain                     1.1.5  Predict rain tomorrow using.                
scatter                  1.2.2  Demonstrate the concept of scatter plots.   

To locally install a model downloaded from an ML Hub:

  $ ml install <model>

fred@dl03:~$ ml install object-recognition 
Model https://mlhub.ai/pool/main/o/object-recognition/object-recognition_1.2.1.mlm

Downloading 'object-recognition_1.2.1.mlm' (227,229,227 bytes) ...

Extracted 'object-recognition_1.2.1.mlm' into
'/home/fred/.mlhub/object-recognition' (244,506,602 bytes).

To display the model's README information:

  $ ml readme object-recognition

fred@dl03:~$ ml demo object-recognition http://hounslowurbanfarm.co.uk/wp-content/uploads/2017/03/img-animal-Willow-the-Barn-Owl.jpg
mlhub: model 'img-animal-Willow-the-Barn-Owl.jpg' is not installed (/home/fred/.mlhub/).

Check for the model name amongst those installed:

  $ ml installed

Models can be installed from the ML Hub:

  $ ml install img-animal-Willow-the-Barn-Owl.jpg

Available pakages on the ML Hub can be listed with:

  $ ml available

fred@dl03:~$ ml --version
1.4.5
```

**Comment by gjwgit** [comment]
And yet, running the same command immediately afterwards works:

```
fred@dl03:~$ ml demo object-recognition http://hounslowurbanfarm.co.uk/wp-content/uploads/2017/03/img-animal-Willow-the-Barn-Owl.jpg
Graphic display is required but not available for command 'demo'. Continue [y/N]? 

```

**Comment by simonzhaoms** [comment]
Uninstall and remove `~/.mlhub`:
```console
$ pip uninstall mlhub
$ rm -rf ~/.mlhub/
```
And then log out and re-login:
```console
$ pip install mlhub
$ ml install object-recognition
$ ml score object-recognition http://hounslowurbanfarm.co.uk/wp-content/uploads/2017/03/img-animal-Willow-the-Barn-Owl.jpg
Namespace(cmd='score', debug=False, func=<function dispatch at 0x7f0a1e9a6730>, init_dir=None, mlhub=None, model='http://hounslowurbanfarm.co.uk/wp-content/uploads/2017/03/img-animal-Willow-the-Barn-Owl.jpg', param=['object-recognition'], quiet=False, version=False)
mlhub: model 'img-animal-Willow-the-Barn-Owl.jpg' is not installed (/home/simon/.mlhub/).
...
$ ml score object-recognition http://hounslowurbanfarm.co.uk/wp-content/uploads/2017/03/img-animal-Willow-the-Barn-Owl.jpg
Namespace(cmd='score', debug=False, func=<function dispatch at 0x7faabe955730>, init_dir=None, mlhub=None, model='object-recognition', param=['http://hounslowurbanfarm.co.uk/wp-content/uploads/2017/03/img-animal-Willow-the-Barn-Owl.jpg'], quiet=False, version=False)
Graphic display is required but not available for command 'score'. Continue [y/N]?
...
```
The `Namespace(...)` in the output is a log printed from `__init__.py` by using `print(args)` for `args = parser.parse_args(sys.argv)`.

You can see different treatment of the url by `argparse`.

**Comment by simonzhaoms** [comment]
Solved by commit mlhubber/mlhub@e65a94ab95eabb7d6af653d5f04c9bff6a2846ac
